src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,419 @@
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static int HexToInt(string value)
+        {
+            // strip the leading 0x
+            if (value.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
+            {
+                value = value.Substring(2);
+            }
+            return Int32.Parse(value, NumberStyles.HexNumber);
+        }
+
+        private static void ParseEntry(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                XmlNode bestPointSoFar = null;
+                int ilOffset = HexToInt(stInfo.Offset);
+                foreach (XmlNode node in xn)
+                {
+                    // If the attribute is not 'startLine', but 'hidden', select the best value so far
+                    if (HexToInt(node.Attributes["offset"].Value) > ilOffset)
+                    {
+                        break;
+                    }
+                    if (node.Attributes["startLine"] != null)
+                    {
+                        bestPointSoFar = node;
+                    }
+                }
+                if (bestPointSoFar != null)
+                {
+                    stInfo.StartLine = bestPointSoFar.Attributes["startLine"].Value;
+                    stInfo.EndLine = bestPointSoFar.Attributes["endLine"].Value;
+                    stInfo.Document = bestPointSoFar.Attributes["document"].Value;
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        public static Command SymbolicateCommand() =>
+            new Command(
+                name: "symbolicate", description: "Get the line number from the Method Token and IL Offset at the stacktrace")
+            {
+                // Handler
+                HandlerDescriptor.FromDelegate((SymbolicateDelegate)Symbolicate).GetCommandHandler(),
+                // Arguments and Options
+                InputArgument(),
+                SearchDirectoryOption(),
+                OutputOption()
+            };
+
+        public static Argument InputArgument() =>
+            new Argument<FileInfo[]>(name: "input-path")
+            {
+                Description = "Path to the stacktrace text file"
+            }.ExistingOnly();
+
+        public static Option SearchDirectoryOption() =>
+            new Option(
+                aliases: new[] { "-d", "--search-dir" },
+                description: "Path of multiple directories with assembly and pdb separated by semicolon(';')")
+            {
+                Argument = new Argument<string>(name: "dir1;dir2;...")
+            };
+
+        public static Option OutputOption() =>
+            new Option(
+                aliases: new[] { "-o", "--output" },
+                description: "Output directly to a file")
+            {
+                Argument = new Argument<string>(name: "output_path")
```

Comment:
This is an outdated version of System.CommandLine. With the new version you'll need to use `Option<T>` and you can't set the `Argument` property.

Reply:
I modified it.